lib/generator.js

```diff
@@ -468,7 +468,17 @@ class Generator {
   async generateSeparateFile(asyncapiDocument, name, component, template, fileName, baseDir) {
     try {
       const relativeBaseDir = path.relative(this.templateContentDir, baseDir);
-      const newFileName = fileName.replace(`\$\$${template}\$\$`, filenamify(name, { replacement: '-', maxLength: 255 }));
+
+      const beforeFileGenerationHookName = 'generate:changeFilename'
+      let filename = name
+      if(this.isHookAvailable(beforeFileGenerationHookName)){
+        let filenamesFromHooks = await this.launchHook(beforeFileGenerationHookName, {"filename": filename})
+        //Use the result of the first hook
+        filename = filenamesFromHooks[0];
```

Comment:
IMO we only need use the first hook found. Should that somehow be stated in the documentation?

Reply:
yes, as it kind of indicates that you should have only one hook that modifies names as we cannot assure which one will be triggered first